Add function to get info of ISO channel

Thalley opened the pull request:
Implements bt_iso_chan_get_info such that the application may get information about a bt_iso_chan object. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/41595

Thalley committed:
Bluetooth: ISO: Add ISO channel type

Add a enum for the ISO channel type instead of using
a boolean for just bis/cis.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Add function to get info of ISO channel

The application may want to want the type of an
ISO channel, and take action based on what the type is.

It has been implemented as a get_info to be
consistent with other get_info functions in the
Bluetooth subsystem.

The bt_iso_info struct can be expanded with more information
later as required.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

koffes reviewed (approved):
Looks great!

cvinayak commented on subsys/bluetooth/host/conn_internal.h:
```diff
@@ -2,6 +2,8 @@
  *  @brief Internal APIs for Bluetooth connection handling.
  */
 
+#include <bluetooth/iso.h>
+
```
We do not include headers in internal header files.

Thalley replied:
Hmm, to avoid including it here, it has to be included by any file that includes `conn_internal.h`, which is about 26 places, almost none of which has anything to do with ISO. 

The only alternative I can think of, is to move `struct bt_conn_iso`, which uses the type from `iso.h`, to the `iso_internal.h` file instead. 

But also, as a rebuttal, we do that in rfcomm_internal.h, l2cap_internal.h, iso_internal.h, avdtp_internal.h as well as the internal headers for LE Audio.

cvinayak replied:
Why `struct bt_conn_iso` is here, wouldn't moving `struct bt_conn_iso`  into iso_internal.h reduce including in iso.h in non-iso related c files?

I am ok to take this outside of this PR as a github issue that can be fixed subsequently, i.e. to avoid including definitions in unrelated c source files.

Thalley replied:
I don't disagree with your observations. I don't know why it was added to conn_internal.h instead of `iso_internal.h`, where it could also have been defined. We can do that in a separate PR. 

Do you think that `struct bt_conn_br` should also be moved to `br.h` then? 

@alwa-nordic @Vudentz You have any input here?

cvinayak commented on subsys/bluetooth/host/iso.c:
```diff
@@ -346,7 +344,7 @@ void bt_iso_connected(struct bt_conn *iso)
 		BT_ERR("Unable to setup data path");
 #if defined(CONFIG_BT_ISO_BROADCAST)
-		if (iso->iso.is_bis) {
+		if (iso->iso.type != BT_ISO_CHAN_TYPE_CONNECTED) {
```
Use explicit check similar to in line 310, and assertion on BT_ISO_CHAN_TYPE_NONE

```suggestion
		if (iso->iso.type == BT_ISO_CHAN_TYPE_BROADCASTER ||
		    iso->iso.type == BT_ISO_CHAN_TYPE_SYNC_RECEIVER) {
```

Thalley replied:
Considered doing that, but didn't as that would change unnecessary many lines, but it is a good idea to do it as it is no longer a boolean value. Will update.

cvinayak commented on subsys/bluetooth/host/iso.c:
```diff
@@ -382,7 +380,7 @@ static void bt_iso_remove_data_path(struct bt_conn *iso)
 	BT_DBG("%p", iso);
 
-	if (iso->iso.is_bis) {
+	if (iso->iso.type != BT_ISO_CHAN_TYPE_CONNECTED) {
```
Use explicit check for each enum value, or have assertion on unlikely values.

cvinayak commented on subsys/bluetooth/host/iso.c:
```diff
@@ -1144,7 +1166,7 @@ static int cig_init_cis(struct bt_iso_cig *cig,
 
 			cis->iso->iso.cig_id = cig->id;
-			cis->iso->iso.is_bis = false;
+			cis->iso->iso.type = BT_ISO_CHAN_TYPE_CONNECTED;
```
Use local variable to reduce number of double dereferencing `bis->iso->iso`

cvinayak commented on subsys/bluetooth/host/iso.c:
```diff
@@ -1681,7 +1703,8 @@ static int big_init_bis(struct bt_iso_big *big,
 
 		bis->iso->iso.big_handle = big->handle;
-		bis->iso->iso.is_bis = true;
+		bis->iso->iso.type = broadcaster ? BT_ISO_CHAN_TYPE_BROADCASTER
+						 : BT_ISO_CHAN_TYPE_SYNC_RECEIVER;
```
Use local variable to reduce number of double dereferencing `bis->iso->iso`

Thalley committed:
Bluetooth: ISO: Remove double dereferencing in init functions

Remove the double dereferencing in the big_init_bis and
cig_init_cis functions.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

cvinayak commented on subsys/bluetooth/host/conn_internal.h:
```diff
@@ -2,6 +2,8 @@
 
+#include <bluetooth/iso.h>
+
 /*
  * Copyright (c) 2015 Intel Corporation
  *
```
Fix copyright, add Nordic copyright line, and have the lines at the top of the file.

cvinayak reviewed (approved):
No changes requested, apart from an open discussion on decoupling ISO declaration from Conn declarations.
